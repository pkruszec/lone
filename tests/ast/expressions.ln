// Unary plus/minus
-4;
+2;
+-4;
--2;
++--56;

// Basic operations
2 + 2;
2 + 2 - 4 + 5 - 4;
2*2 + 4*4 + 5*5;
5 / (3 + 2);
5 / 3 + 2;
2 / 3 % 4;
2 / (3 % 4);

// Casts
cast(int) 5.4;
cast(float) 5;
cast(float)(cast(int) 3.3);
