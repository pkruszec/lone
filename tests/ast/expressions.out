PROG
  EVAL
    U_MINUS
      NUM(4)
  EVAL
    U_PLUS
      NUM(2)
  EVAL
    U_PLUS
      U_MINUS
        NUM(4)
  EVAL
    U_MINUS
      U_MINUS
        NUM(2)
  EVAL
    U_PLUS
      U_PLUS
        U_MINUS
          U_MINUS
            NUM(56)
  EVAL
    ADD
      NUM(2)
      NUM(2)
  EVAL
    SUB
      ADD
        SUB
          ADD
            NUM(2)
            NUM(2)
          NUM(4)
        NUM(5)
      NUM(4)
  EVAL
    ADD
      ADD
        MUL
          NUM(2)
          NUM(2)
        MUL
          NUM(4)
          NUM(4)
      MUL
        NUM(5)
        NUM(5)
  EVAL
    DIV
      NUM(5)
      ADD
        NUM(3)
        NUM(2)
  EVAL
    ADD
      DIV
        NUM(5)
        NUM(3)
      NUM(2)
  EVAL
    MOD
      DIV
        NUM(2)
        NUM(3)
      NUM(4)
  EVAL
    DIV
      NUM(2)
      MOD
        NUM(3)
        NUM(4)
  EVAL
    CAST
      IDENT(int)
      NUM(5.4)
  EVAL
    CAST
      IDENT(float)
      NUM(5)
  EVAL
    CAST
      IDENT(float)
      CAST
        IDENT(int)
        NUM(3.3)
